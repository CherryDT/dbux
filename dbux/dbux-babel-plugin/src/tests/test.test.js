import pluginTester from 'babel-plugin-tester';
import plugin from '..';
import path from 'path';

//import { promises as fs } from 'fs';
import fs from 'fs';

const codeFolder = path.join(__dirname, '__code__');
const fpath = path.join(codeFolder, 'example/code.js');

let code;

// beforeAll(() => {
  code = fs.readFileSync(fpath, { encoding: 'utf8' });

  expect(code).toBeString();
// });


pluginTester({
  plugin,
  pluginName: 'dbux-babel-plugin',
  babelOptions: {
    configFile: path.join(__dirname, '../../babel.config.js'),
  },
  filename: __filename,
  tests: [
    {
      code,
      snapshot: true
    }
  ]
});