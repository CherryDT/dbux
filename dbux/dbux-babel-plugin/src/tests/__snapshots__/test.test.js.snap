// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dbux-babel-plugin example1.js: example1.js 1`] = `

function f1(x,y) {
}

class A {
  constructor() {
    this.m = new Map();
    this.s = new Set();
  }

  getX() {}
}

const f2 = (z,w) => {};

export {
  f1,
  A,
  f2
}

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.map.js");

require("core-js/modules/es.object.to-string.js");

require("core-js/modules/es.set.js");

require("core-js/modules/es.string.iterator.js");

require("core-js/modules/web.dom-collections.iterator.js");

exports.__esModule = true;
exports.f1 = f1;
exports.f2 = exports.A = void 0;

function f1(x, y) {
  "before";
  "after";
}

var A =
  /*#__PURE__*/
  (function() {
    function A() {
      "before";
      this.m = new Map();
      this.s = new Set();
      ("after");
    }

    var _proto = A.prototype;

    _proto.getX = function getX() {
      "before";
      "after";
    };

    return A;
  })();

exports.A = A;

var f2 = function f2(z, w) {
  "before";
  "after";
};

exports.f2 = f2;


`;
